Update where app-model actions/providers/processors live in codebase (#416)
napari/docs: docs/developers/architecture/app_model.md

### `Action`s in napari

In napari, menu bar actions are defined in
[`napari/_qt/_qapp_model/qactions`](https://github.com/napari/napari/tree/main/napari/_qt/_qapp_model/qactions),
with one file per menu.
While not all menu bar actions strictly require Qt, they are defined for the purpose

> of living in a menu and are thus considered 'Qt' actions.

Reviewer: ```suggestion
of living in a menu and are thus considered GUI actions. If we move to supporting more GUI backends, we may want to move these out of `_qt` to a shared `gui` folder, depending on implementation.
```

> This also ensures that there is only one file defining.

Reviewer: ```suggestion
This also ensures that there is only one file defining both actions and their submenus.
```

> The layer context menu actions do not require a GUI and thus live in

Reviewer: ```suggestion
The layer context menu actions do not require a GUI (they only require the `layerlist`) and thus live in
```
